pkg/controllers/static/deprovisioning/controller.go

```diff
@@ -174,11 +173,72 @@ func HasNodePoolReplicaCountChanged(oldNP, newNP *v1.NodePool) bool {
 	return lo.FromPtr(oldNP.Spec.Replicas) != lo.FromPtr(newNP.Spec.Replicas)
 }
 
-// Returns nodes suitable for deprovisioning, prioritizing:
-// 1. Empty nodes (nodes with no pods or only DaemonSet pods without do-not-disrupt annotation)
-// 2. If more nodes needed, nodes with lowest disruption cost (nodes with pods that have do-not-disrupt will have highest cost)
-func (c *Controller) getDeprovisioningCandidates(ctx context.Context, np *v1.NodePool, nodes []*state.StateNode, count int) []*state.StateNode {
-	// First get empty nodes
+// Returns NodeClaims suitable for deprovisioning, prioritizing:
+// 1. Unresolved NodeClaims (no ProviderID yet - haven't launched)
+// 2. Empty nodes (nodes with no pods or only DaemonSet pods without do-not-disrupt annotation)
+// 3. If more nodes needed, nodes with lowest disruption cost (nodes with pods that have do-not-disrupt will have highest cost)
+func (c *Controller) getDeprovisioningCandidates(ctx context.Context, np *v1.NodePool, count int) []*v1.NodeClaim {
+	candidates := make([]*v1.NodeClaim, 0, count)
+
+	// Unresolved NodeClaims (haven't launched yet or that failed CreateFleet call)
```

Comment:
I wouldn't directly call out CreateFleet in the upstream codebase. It's sufficient to just say that the node hasn't been successfully launched yet.